builddep command doesn't support --allowerasing

SSIA, this option has been used by Mock for some time.

Hey @m-blaha, sorry if I follow up 3 days after your message but I've been busy with school and work this week. 

I am not familiar with ccache and the various C/C++ tools as I don't code with these languages often. Sorry. :P

By the way, I used your solution and it works, but this time an other error showed up:

```
[ 57%] Built target python3_progressbar
[ 57%] Building CXX object bindings/ruby/libdnf5/CMakeFiles/ruby_advisory.dir/CMakeFiles/ruby_advisory.dir/advisoryRUBY_wrap.cxx.o
[ 57%] Building CXX object bindings/ruby/libdnf5/CMakeFiles/ruby_base.dir/CMakeFiles/ruby_base.dir/baseRUBY_wrap.cxx.o
In file included from /home/bavuett/.rbenv/versions/2.7.6/include/ruby-2.7.0/ruby/intern.h:39,
                 from /home/bavuett/.rbenv/versions/2.7.6/include/ruby-2.7.0/ruby/ruby.h:2150,
                 from /home/bavuett/.rbenv/versions/2.7.6/include/ruby-2.7.0/ruby.h:33,
                 from /home/bavuett/Coding/GitHub/dnf5/build/bindings/ruby/libdnf5/CMakeFiles/ruby_advisory.dir/advisoryRUBY_wrap.cxx:872:
/home/bavuett/.rbenv/versions/2.7.6/include/ruby-2.7.0/ruby/st.h:168:42: error: redundant redeclaration of ‘int rb_st_locale_insensitive_strcasecmp(const char*, const char*)’ in same scope [-Werror=redundant-decls]
  168 | #define st_locale_insensitive_strcasecmp rb_st_locale_insensitive_strcasecmp
      |                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/bavuett/.rbenv/versions/2.7.6/include/ruby-2.7.0/ruby/ruby.h:2323:5: note: in expansion of macro ‘st_locale_insensitive_strcasecmp’
 2323 | int st_locale_insensitive_strcasecmp(const char *s1, const char *s2);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/bavuett/.rbenv/versions/2.7.6/include/ruby-2.7.0/ruby/ruby.h:24:
/home/bavuett/.rbenv/versions/2.7.6/include/ruby-2.7.0/ruby/st.h:167:14: note: previous declaration of ‘int rb_st_locale_insensitive_strcasecmp(const char*, const char*)’
  167 | PUREFUNC(int rb_st_locale_insensitive_strcasecmp(const char *s1, const char *s2));
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/bavuett/.rbenv/versions/2.7.6/include/ruby-2.7.0/x86_64-linux/ruby/config.h:140:48: note: in definition of macro ‘PUREFUNC’
  140 | #define PUREFUNC(x) __attribute__ ((__pure__)) x
      |                                                ^
/home/bavuett/.rbenv/versions/2.7.6/include/ruby-2.7.0/ruby/st.h:170:43: error: redundant redeclaration of ‘int rb_st_locale_insensitive_strncasecmp(const char*, const char*, size_t)’ in same scope [-Werror=redundant-decls]
  170 | #define st_locale_insensitive_strncasecmp rb_st_locale_insensitive_strncasecmp
      |                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/bavuett/.rbenv/versions/2.7.6/include/ruby-2.7.0/ruby/ruby.h:2324:5: note: in expansion of macro ‘st_locale_insensitive_strncasecmp’
 2324 | int st_locale_insensitive_strncasecmp(const char *s1, const char *s2, size_t n);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/bavuett/.rbenv/versions/2.7.6/include/ruby-2.7.0/ruby/st.h:169:14: note: previous declaration of ‘int rb_st_locale_insensitive_strncasecmp(const char*, const char*, size_t)’
  169 | PUREFUNC(int rb_st_locale_insensitive_strncasecmp(const char *s1, const char *s2, size_t n));
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/bavuett/.rbenv/versions/2.7.6/include/ruby-2.7.0/x86_64-linux/ruby/config.h:140:48: note: in definition of macro ‘PUREFUNC’
  140 | #define PUREFUNC(x) __attribute__ ((__pure__)) x
      |                                                ^
cc1plus: note: unrecognized command-line option ‘-Wno-sometimes-uninitialized’ may have been intended to silence earlier diagnostics
cc1plus: all warnings being treated as errors
make[2]: *** [bindings/ruby/libdnf5/CMakeFiles/ruby_advisory.dir/build.make:76: bindings/ruby/libdnf5/CMakeFiles/ruby_advisory.dir/CMakeFiles/ruby_advisory.dir/advisoryRUBY_wrap.cxx.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:2721: bindings/ruby/libdnf5/CMakeFiles/ruby_advisory.dir/all] Error 2
make[1]: *** Attesa per i processi non terminati....
[ 57%] Built target perl5_rpm
In file included from /home/bavuett/.rbenv/versions/2.7.6/include/ruby-2.7.0/ruby/intern.h:39,
                 from /home/bavuett/.rbenv/versions/2.7.6/include/ruby-2.7.0/ruby/ruby.h:2150,
                 from /home/bavuett/.rbenv/versions/2.7.6/include/ruby-2.7.0/ruby.h:33,
                 from /home/bavuett/Coding/GitHub/dnf5/build/bindings/ruby/libdnf5/CMakeFiles/ruby_base.dir/baseRUBY_wrap.cxx:873:
/home/bavuett/.rbenv/versions/2.7.6/include/ruby-2.7.0/ruby/st.h:168:42: error: redundant redeclaration of ‘int rb_st_locale_insensitive_strcasecmp(const char*, const char*)’ in same scope [-Werror=redundant-decls]
  168 | #define st_locale_insensitive_strcasecmp rb_st_locale_insensitive_strcasecmp
      |                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/bavuett/.rbenv/versions/2.7.6/include/ruby-2.7.0/ruby/ruby.h:2323:5: note: in expansion of macro ‘st_locale_insensitive_strcasecmp’
 2323 | int st_locale_insensitive_strcasecmp(const char *s1, const char *s2);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/bavuett/.rbenv/versions/2.7.6/include/ruby-2.7.0/ruby/ruby.h:24:
/home/bavuett/.rbenv/versions/2.7.6/include/ruby-2.7.0/ruby/st.h:167:14: note: previous declaration of ‘int rb_st_locale_insensitive_strcasecmp(const char*, const char*)’
  167 | PUREFUNC(int rb_st_locale_insensitive_strcasecmp(const char *s1, const char *s2));
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/bavuett/.rbenv/versions/2.7.6/include/ruby-2.7.0/x86_64-linux/ruby/config.h:140:48: note: in definition of macro ‘PUREFUNC’
  140 | #define PUREFUNC(x) __attribute__ ((__pure__)) x
      |                                                ^
/home/bavuett/.rbenv/versions/2.7.6/include/ruby-2.7.0/ruby/st.h:170:43: error: redundant redeclaration of ‘int rb_st_locale_insensitive_strncasecmp(const char*, const char*, size_t)’ in same scope [-Werror=redundant-decls]
  170 | #define st_locale_insensitive_strncasecmp rb_st_locale_insensitive_strncasecmp
      |                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/bavuett/.rbenv/versions/2.7.6/include/ruby-2.7.0/ruby/ruby.h:2324:5: note: in expansion of macro ‘st_locale_insensitive_strncasecmp’
 2324 | int st_locale_insensitive_strncasecmp(const char *s1, const char *s2, size_t n);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/bavuett/.rbenv/versions/2.7.6/include/ruby-2.7.0/ruby/st.h:169:14: note: previous declaration of ‘int rb_st_locale_insensitive_strncasecmp(const char*, const char*, size_t)’
  169 | PUREFUNC(int rb_st_locale_insensitive_strncasecmp(const char *s1, const char *s2, size_t n));
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/bavuett/.rbenv/versions/2.7.6/include/ruby-2.7.0/x86_64-linux/ruby/config.h:140:48: note: in definition of macro ‘PUREFUNC’
  140 | #define PUREFUNC(x) __attribute__ ((__pure__)) x
      |                                                ^
cc1plus: note: unrecognized command-line option ‘-Wno-sometimes-uninitialized’ may have been intended to silence earlier diagnostics
cc1plus: all warnings being treated as errors
make[2]: *** [bindings/ruby/libdnf5/CMakeFiles/ruby_base.dir/build.make:76: bindings/ruby/libdnf5/CMakeFiles/ruby_base.dir/CMakeFiles/ruby_base.dir/baseRUBY_wrap.cxx.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:2775: bindings/ruby/libdnf5/CMakeFiles/ruby_base.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

I've tried looking up online before bothering you again, but I couldn't find anything useful. Adding the `-i` parameter to `make` solves this (since the error seems to be only caused by the Ruby bindings), but I don't like this approach either so I figured it would be useful to report it to you. Have a nice weekend.

--- a/dnf5-plugins/builddep_plugin/builddep.cpp
+++ b/dnf5-plugins/builddep_plugin/builddep.cpp
@@ -59,6 +59,8 @@ void BuildDepCommand::set_argument_parser() {
     });
     cmd.register_positional_arg(specs);
 
+    allow_erasing = std::make_unique<AllowErasingOption>(*this);
+
     auto defs = parser.add_new_named_arg("rpm_macros");
     defs->set_short_name('D');
     defs->set_long_name("define");
@@ -247,6 +249,8 @@ void BuildDepCommand::run() {
 
     // fill the goal with build dependencies
     auto goal = get_context().get_goal();
+    goal->set_allow_erasing(allow_erasing->get_value());
+    
     for (const auto & spec : install_specs) {
         if (libdnf::rpm::Reldep::is_rich_dependency(spec)) {
             goal->add_provide_install(spec);

--- a/dnf5-plugins/builddep_plugin/builddep.cpp
+++ b/dnf5-plugins/builddep_plugin/builddep.cpp
@@ -59,8 +59,6 @@ void BuildDepCommand::set_argument_parser() {
     });
     cmd.register_positional_arg(specs);
 
-    allow_erasing = std::make_unique<AllowErasingOption>(*this);
-
     auto defs = parser.add_new_named_arg("rpm_macros");
     defs->set_short_name('D');
     defs->set_long_name("define");
@@ -82,6 +80,7 @@ void BuildDepCommand::set_argument_parser() {
         });
     cmd.register_named_arg(defs);
 
+    auto allow_erasing = std::make_unique<AllowErasingOption>(*this);
     auto skip_unavailable = std::make_unique<SkipUnavailableOption>(*this);
 }
 

--- a/dnf5-plugins/builddep_plugin/builddep.cpp
+++ b/dnf5-plugins/builddep_plugin/builddep.cpp
@@ -59,6 +59,9 @@ void BuildDepCommand::set_argument_parser() {
     });
     cmd.register_positional_arg(specs);
 
+    allow_erasing = std::make_unique<AllowErasingOption>(*this);
+    auto skip_unavailable = std::make_unique<SkipUnavailableOption>(*this);
+
     auto defs = parser.add_new_named_arg("rpm_macros");
     defs->set_short_name('D');
     defs->set_long_name("define");
@@ -79,9 +82,6 @@ void BuildDepCommand::set_argument_parser() {
             return true;
         });
     cmd.register_named_arg(defs);
-
-    auto allow_erasing = std::make_unique<AllowErasingOption>(*this);
-    auto skip_unavailable = std::make_unique<SkipUnavailableOption>(*this);
 }
 
 void BuildDepCommand::configure() {

--- a/dnf5-plugins/builddep_plugin/builddep.hpp
+++ b/dnf5-plugins/builddep_plugin/builddep.hpp
@@ -23,6 +23,7 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 
 
 #include <dnf5/context.hpp>
+#include <dnf5/shared_options.hpp>
 #include <libdnf/conf/option_bool.hpp>
 
 #include <vector>
@@ -53,6 +54,8 @@ private:
     std::vector<std::string> spec_file_paths{};
     std::vector<std::string> srpm_file_paths{};
     std::vector<std::pair<std::string, std::string>> rpm_macros{};
+
+    std::unique_ptr<AllowErasingOption> allow_erasing;
 };
 
 

